### DISCUSSION TITLE
Erhöhen der Länge des Feldes "scriptPath" in der Tabelle "task"

### DISCUSSION THREAD

**Discussion by gerlingsm** [kitodo/kitodo-production#6108] [Ideas]
In einem früheren [Merge-Request](https://github.com/kitodo/kitodo-production/pull/1740) ist die maximale Länge von scriptPath bereits einmal auf 500 erhöht worden.
Dieser Wert ist für die von mir betreute Umgebung zu wenig, da auch Scriptaufrufe mit einer Länge von über 500 Zeichen vorkommen.
Ich hätte eine allgemeine Erhöhung auf 2000 (oder mehr) vorgeschlagen um erst einmal ein wenig Platz nach oben zu haben.
Da ich aber nicht genau weiß, ob dies ggf an anderer Stelle negative Auswirkungen hat erst einmal hier als Idee.
Falls gewünscht kann ich aber auch noch einen Merge-Request erstellen.
Für die eigene Datenbank habe ich dies bereits mit dem Befehl `ALTER TABLE task MODIFY COLUMN scriptPath VARCHAR(2000);` auf 2000 erhöht.

**Comment by henning-gerhardt** [comment]
MariaDB unterstützt beim Datentyp VARCHAR bis zu 65.532 Zeichen (ASCII) bzw. bei UTF8 bis zu 21.844 Zeichen (siehe https://mariadb.com/kb/en/varchar/), daher sollte die Vergrößerung technisch gesehen kein Problem darstellen. Ggf. wird die Datenbank-Tabelle `task` dadurch größer, aber wie stark sich das auswirkt, ist mir nicht bekannt.

Mich wundert es eher, warum die Script-Aufrufe so lang sind, da schon 500 Zeichen in der Regel ausreichend sein sollten.

**Reply by gerlingsm** [reply]
In dem genannten Szenario werden Metadaten direkt an die Scripte übergeben beispielsweise so:
/usr/local/bin/subfolder/describing_scriptname.sh --kitodo_processid (processid) --kitodo_processtitle (processtitle) --kitodo_stepname (stepname) --meta_document_type ${meta.document_type} ...
Da es sich um viele Metadaten handelt, welche an die Scripte übergeben werden, sind wir an das Limit gestoßen.
